src/app/components/transactionSetting/feeItem.tsx

```diff
@@ -0,0 +1,121 @@
+import { Bicycle, CarProfile, RocketLaunch } from '@phosphor-icons/react';
+import { StyledP } from '@ui-library/common.styled';
+import styled from 'styled-components';
+import Theme from 'theme';
+
+interface FeeContainer {
+  isSelected: boolean;
+}
+
+const FeeItemContainer = styled.button<FeeContainer>`
+  display: flex;
+  padding: 12px 16px;
+  align-items: center;
+  gap: 12px;
+  align-self: stretch;
+  border-radius: 12px;
+  border: 1px solid var(--White-850, rgba(255, 255, 255, 0.15));
+  flex-direction: row;
+  background: ${(props) => (props.isSelected ? props.theme.colors.elevation6_600 : 'transparent')};
+  margin-top: ${(props) => props.theme.space.xs};
+  flex: 1;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const TextsContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+
+  flex: 1;
+`;
+
+const ColumnsTexts = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  flex: 1;
+`;
+const EndColumnTexts = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+`;
+
+const StyledHeading = styled(StyledP)`
+  margin-bottom: ${(props) => props.theme.space.xxs};
+`;
+
+const StyledSubText = styled(StyledP)`
+  color: ${(props) => props.theme.colors.white_200};
+  margin-bottom: ${(props) => props.theme.space.xxs};
+`;
+
+type FeePriority = 'high' | 'medium' | 'low';
+
+interface FeeItemProps {
+  priority: FeePriority;
+  time: string;
+  feeRate: string;
+  totalFee: string;
+  fiat: string | JSX.Element;
+  selected: boolean;
+  onClick?: () => void;
+}
+
+function FeeItem({ priority, time, feeRate, totalFee, fiat, selected, onClick }: FeeItemProps) {
+  const getIcon = () => {
+    switch (priority) {
+      case 'high':
+        return <RocketLaunch size={20} color={Theme.colors.tangerine} />;
+      case 'medium':
+        return <CarProfile size={20} color={Theme.colors.tangerine} />;
+      case 'low':
+        return <Bicycle size={20} color={Theme.colors.tangerine} />;
+      default:
+        return <RocketLaunch size={20} color={Theme.colors.tangerine} />;
+    }
+  };
+
+  // todo: localisation
```

Comment:
I think we can use the localization keys from the `SPEED_UP_TRANSACTION` object (`src/locales/en.json`) here for now